base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
+"my/test/dir/"
 ```
 """
 function mkpath(path::AbstractString; mode::Integer = 0o777)
-    isdirpath(path) && (path = dirname(path))
     dir = dirname(path)
+    # stop recursion for `""`, `"/"`, or existed dir
     (path == dir || isdir(path)) && return path
     mkpath(dir, mode = checkmode(mode))
```

Comment:
I think what would make this code easier to understand to me would be simply renaming `dir` to e.g. `parentdir`. Once I got that everything else made sense